### PR TITLE
Enable Embree build with SYCL support and OneAPI 2024.1.0

### PR DESCRIPTION
**Pull request by lumurillo**
Enable the SYCL support for the Embree package.

## Type

-   [ ] Bug fix (non-breaking change which fixes an issue): Fixes #
-   [x] New feature (non-breaking change which adds functionality). Resolves #
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) Resolves #

## Motivation and Context

This change will allow the implementation of SYCL GPU implementation for the `RayCastingScene` class.

## Checklist:

-   [x] I have run `python util/check_style.py --apply` to apply Open3D **code style**
    to my code.
-   [x] This PR changes Open3D behavior or adds new functionality.
    -   [ ] Both C++ (Doxygen) and Python (Sphinx / Google style) **documentation** is
        updated accordingly.
    -   [ ] I have added or updated C++ and / or Python **unit tests** OR included **test
        results** (e.g. screenshots or numbers) here.
-   [x] I will follow up and update the code if CI fails.
    
-   [x] For fork PRs, I have selected **Allow edits from maintainers**.

## Description

* Updated the OneAPI version for the Docker files.
* Updated the CMake version.
* Disable the build of the ISPC module.

### CODE REVIEW

**Comment by ssheorey** [comment]
Hi @lumurillo can you update the embree Python examples to use SYCL devices, if available?

```
examples/python/geometry/ray_casting_closest_geometry.py
examples/python/geometry/ray_casting_sdf.py
examples/python/geometry/ray_casting_to_image.py
```

File: cpp/open3d/t/geometry/RaycastingScene.h [hunk 1]
```diff
@@ -30,7 +30,11 @@ namespace geometry {
 class RaycastingScene {
 public:
     /// \brief Default Constructor.
-    RaycastingScene(int64_t nthreads = 0);
+    RaycastingScene(int64_t nthreads = 0
+#ifdef BUILD_SYCL_MODULE
+        , const core::Device& device = core::Device("CPU:0")
+#endif
```
[the remark this answers is not in the document]
Comment (lumurillo) [reply cpp/open3d/t/geometry/RaycastingScene.h]: See 6284e3d25d5e620cfa0cb135d1de1cfcb71fccc2

File: cpp/open3d/t/geometry/RaycastingScene.cpp [hunk 2]
```diff
@@ -689,16 +1169,50 @@ struct RaycastingScene::Impl {
                     LoopFn);
         }
     }
+
```
Comment (ssheorey) [inline cpp/open3d/t/geometry/RaycastingScene.cpp:1172]: Can we use STL here for ArraySum and ArrayPartialSum, and memcpy for CopyArray?

**Review by ssheorey — COMMENTED** [review]
Hi Luis, looks good! See some review comments below.

[Performance] About `queue.wait_and_throw()` - I would recommend avoiding it in each function. Instead have a sycl utils level `wait_and_throw()`. This lets users launch a raycast task on the GPU, do something else on the CPU (maybe read more data, etc.) then do a `queue.wait_and_throw()` to get results. 

The current alternative would be multithreading (check if GIL is released for raycast ops in Python), but this needs quite a bit more effort from the user.

File: cpp/open3d/t/geometry/RaycastingScene.cpp [hunk 3]
```diff
@@ -158,27 +137,75 @@ void ListIntersectionsFunc(const RTCFilterFunctionNArguments* args) {
...
+} // namespace callbacks
+
+namespace {
+
+typedef Eigen::AlignedVector3<float> Vec3fa;
+// Dont force alignment for Vec2f because we use it just for storing
+typedef Eigen::Matrix<float, 2, 1, Eigen::DontAlign> Vec2f;
+typedef Eigen::Vector3f Vec3f;
+
+void enablePersistentJITCache()
```
[the remark this answers is not in the document]
Comment (lumurillo) [reply cpp/open3d/t/geometry/RaycastingScene.cpp]: 460e3f7e748c6a2fab5c0db330bb8643885eca5a

File: cpp/open3d/t/geometry/RaycastingScene.cpp [hunk 4]
```diff
@@ -158,27 +137,75 @@ void ListIntersectionsFunc(const RTCFilterFunctionNArguments* args) {
...
+{
+#if defined(_WIN32)
+    _putenv_s("SYCL_CACHE_PERSISTENT","1");
+    _putenv_s("SYCL_CACHE_DIR","cache");
+#else
+    setenv("SYCL_CACHE_PERSISTENT","1",1);
+    setenv("SYCL_CACHE_DIR","cache",1);
+#endif
```
[the remark this answers is not in the document]
Comment (lumurillo) [reply cpp/open3d/t/geometry/RaycastingScene.cpp]: 8da0afce79c9a116a0645917941fa1d558492698

File: python/test/t/geometry/test_raycasting_scene.py [hunk 5]
```diff
@@ -122,14 +122,10 @@ def test_test_lots_of_occlusions(device):
 def test_add_triangle_mesh(device):
     cube = o3d.t.geometry.TriangleMesh.from_legacy(
```
Comment (ssheorey) [inline python/test/t/geometry/test_raycasting_scene.py:123]: Can remove `from_legacy` for mesh creation functions now.

**Comment by ssheorey** [comment]
Some updates needed to pass the CUDA CI tests. I made a branch copy and ran them here:
https://github.com/isl-org/Open3D/actions/runs/11887514382

@lumurillo I'll take care of this.

File: cpp/open3d/t/geometry/RaycastingScene.cpp (hunk 2 above)
Comment (lumurillo) [reply cpp/open3d/t/geometry/RaycastingScene.cpp:1174]: b2689cbe1d0b26c0cccef23bc355d3429bbd4692

File: cpp/open3d/t/geometry/RaycastingScene.cpp [hunk 6]
```diff
@@ -350,15 +390,448 @@ struct RaycastingScene::Impl {
...
+                        rh.hit.geomID = RTC_INVALID_GEOMETRY_ID;
+                        rh.hit.instID[0] = RTC_INVALID_GEOMETRY_ID;
+
+                        rtcIntersect1(scene, &rh, &args);
+                    });
+        });
+        queue_.wait_and_throw();
+
+        // Free the allocated memory
+        sycl::free(previous_geom_prim_ID_tfar, queue_);
```
[the remark this answers is not in the document]
Comment (lumurillo) [reply cpp/open3d/t/geometry/RaycastingScene.cpp]: 4b81b70cfa649d3517a1cf7728fe857abeaed4d0

**Review by ssheorey — APPROVED** [review]

**Comment by ssheorey** [comment]
Looks good. We can visit the `wait_and_throw()` optimization later. Thanks @lumurillo !

**Review by benjaminum — APPROVED** [review]
lgtm

**Pull request merged**